src/backend/distributed/planner/recursive_planning.c

```diff
@@ -1045,12 +1047,12 @@ RecursivelyPlanSetOperations(Query *query, Node *node,
 
 
 /*
- * IsLocalTableRTE gets a node and returns true if the node
+ * IsLocalTableRTEOrMatView gets a node and returns true if the node
  * is a range table relation entry that points to a local
  * relation (i.e., not a distributed relation).
  */
 static bool
-IsLocalTableRTE(Node *node)
+IsLocalTableRTEOrMatView(Node *node)
```

Comment:
What part returns true for MatView? The comments might need to be updated here

Reply:
Do we have tests with mat views?

Reply:
>What part returns true for MatView? The comments might need to be updated here

Here we don't check for mat views to return false and we return true if the underlying relation is a local table (a bit tricky, I know)
When I added mat.view check to there (to return false), some tests started failing so I justified that it actually returns true for mat views pointing to local tables.
The fact that we explicitly check for "views" is another thing behind my justification.


>Do we have tests with mat views?

https://github.com/citusdata/citus/pull/4133/files#diff-0d625087fba32b08457f12fb4dda77d2R802-R813